python/lib/communication/dmod/communication/metadata_message.py

```diff
@@ -25,32 +28,53 @@ def get_value_for_name(cls, name_str: str) -> Optional['MetadataPurpose']:
         return None
 
 
-class MetadataMessage(AbstractInitRequest):
+class MetadataSignal(Serializable):
+    purpose: MetadataPurpose
+    metadata_follows: bool = False
 
-    event_type: MessageEventType = MessageEventType.METADATA
+    class Config:
+        fields = {
+            "metadata_follows": {
```

Comment:
Yes! And thanks for asking. This is an oddity with aliases in `pydantic`. Aliases created using `Field` are not propagated to subtypes _that redeclare the field from the super type_)(i.e. `Field(..., alias="additional_metadata")`). However, aliases specified in the [`fields` model options](https://docs.pydantic.dev/latest/usage/model_config/#options) do propagate to subclasses.

This is fine, `B` does not re-declare `field`.

```python
from pydantic import BaseModel, Field

class A(BaseModel):
    field: str = Field(..., alias="FIELD")

class B(A):
    ...

def test_it():
    # this is fine
    m = B(FIELD="value")
```

This will throw a `ValidationError` because `field` is re-declared.

```python
from pydantic import BaseModel, Field

class A(BaseModel):
    field: str = Field(..., alias="FIELD")

class B(A):
    field: str

def test_it():
    # throws ValidationError
    m = B(FIELD="value")
```

this is fine though ¯\_(ツ)_/¯

```python
from pydantic import BaseModel, Field

class A(BaseModel):
    field: str = Field(description="some field")

    class Config(BaseModel.Config):
        fields = {"field": {"alias": "FIELD"}}

class B(A):
    field: str

def test_it():
    m = B(FIELD="value")
```

Reply:
And we'd run into this with a model that's a composition of two others that both have the `data` field used to represent a message body. No es bueno. Thanks for the clarity!